### PR TITLE
Replaces 'sentinel' with 'policy_enforcement' in FeatureSet and EntitlementSet docs

### PR DESCRIPTION
**Pull request by mfhodges**
### What

Updated example API responses for FeatureSets objects to replace attr `sentinel` with `policy_enforcement`.
Also updated [Feature Entitlements](https://github.com/hashicorp/terraform-docs-common/pull/233/files#diff-88bd17bedafed31edeae3621396512e5449637d6925ed36b6c3f53b6f4da4d06) description with attribute rename.

Changed pages:
- terraform/cloud-docs/api-docs
- terraform/cloud-docs/api-docs/feature-sets
- /terraform/cloud-docs/api-docs/organizations
- /terraform/cloud-docs/api-docs/subscriptions

### Why

https://github.com/hashicorp/atlas/pull/14244 renames `sentinel` attr to `policy_enforcement` for FeatureSets and EntitlementSets.

### Screenshots

----------

### Merge Checklist
_If items do not apply to your changes, add (N/A) and mark them as complete._

#### Pull Request
- [ ] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. tfc).
- [ ] Description links to related pull requests or issues, if any.

#### Content
- [ ] Redirects have been added for moved, renamed, or deleted pages. This requires a separate PR in the [`terraform-website` repository](https://github.com/hashicorp/terraform-website) `redirects.next.js` file.
- [ ] API documentation and the API Changelog have been updated. 
- [ ] Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [ ] Pages with related content are updated and link to this content when appropriate.
- [ ] Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages.
- [ ] New pages have metadata (page name and description) at the top.
- [ ] New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility.
- [ ] New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars.
- [ ] UI elements (button names, page names, etc.) are bolded.
- [ ] The Vercel website preview successfully deployed.

#### Reviews
- [ ] I or someone else reviewed the content for technical accuracy.
- [ ] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

### CODE REVIEW

**Review by judithpatudith — APPROVED** [review]
Thank you for this update! Looks good! 

Unrelated to the docs changes themselves, but I have a kind of naive formatting comment: I noticed that `policy_enforcement` uses underscores and all its sisters use dashes. Is there a reason for that?

**Comment by mfhodges** [comment]
> Thank you for this update! Looks good!
> 
> Unrelated to the docs changes themselves, but I have a kind of naive formatting comment: I noticed that `policy_enforcement` uses underscores and all its sisters use dashes. Is there a reason for that?

🏆 excellent catch!! that is definitely a copy-paste error on my part and I'll update my commit to use dashes shortly. Thanks for the close review 😄

**Review by judithpatudith — APPROVED** [review]

**Comment by mfhodges** [comment]
@judithpatudith I got some feedback on my other PR that we should be keep the attribute `sentinel` in API responses and denote that it is deprecated in the API docs. I'm not sure of any examples of this in the docs so if there's a different pattern I should follow lmk :)

File: website/docs/cloud-docs/api-docs/feature-sets.mdx [hunk 1]
```diff
@@ -259,7 +265,8 @@ curl \
         "plan": "plan_G13T64ifEk3z92",
         "self-serve-billing": true,
         "cost-estimation": true,
-        "sentinel": true,
+        "policy-enforcement": true,
+        "sentinel": false,
```
Comment (mfhodges) [inline website/docs/cloud-docs/api-docs/feature-sets.mdx:269]: 👀 these two fields should be the same

File: website/docs/cloud-docs/api-docs/index.mdx [hunk 2]
```diff
@@ -87,7 +87,7 @@ The following entitlements are available:
 - `run-tasks` — Allows an organization to use [run tasks](/cloud-docs/workspaces/settings/run-tasks). Affects the [run tasks][] endpoints.
 - `self-serve-billing` — Allows an organization to pay via credit card using the in-app billing UI.
 - `state-storage` — Allows an organization to store state versions in its workspaces, which enables local Terraform runs with Terraform Cloud. Affects the [state versions][] endpoints.
-- `sentinel` — Allows an organization to use [Sentinel][]. Affects the [policies][], [policy sets][], and [policy checks][] endpoints.
```
Comment (mfhodges) [inline website/docs/cloud-docs/api-docs/index.mdx:90]: 👀 add sentinel description back

**Pull request merged**